CDCgov/prime-reportstream
# OBR hl7v2 to fhir mapping updates

This PR updates the Service Request and Diagnostic Report resources from the HL7 v2 OBR segment using the latest HL7 v2 to FHIR mapping documents. It also added some mapping between OBR and the Specimen resource.

[Updated mapping documents for OBR](https://cdc.sharepoint.com/:x:/r/teams/USDSatCDC/_layouts/15/Doc.aspx?sourcedoc=%7B3FE6736B-8FD5-48D0-8908-F2D2E323B952%7D&file=HL7%20to%20FHIR%20-%20OBR%20Segment.xlsx&action=default&mobileredirect=true&cid=e4c27b8f-bd2e-43f6-a5c5-877a9941c36a), copy of file attached:
[HL7 to FHIR - OBR Segment.xlsx](https://github.com/CDCgov/prime-reportstream/files/10460810/HL7.to.FHIR.-.OBR.Segment.xlsx)



Test Steps:

1. Build this branch and run the unit and integration tests. Verify they all pass. Alternatively you can verify the Github Actions build passes. Note the HL7 to FHIR to HL7 integration test data was not changed, so the result of our HL7 to FHIR to HL7 conversion remains the same = no data changes.

## Changes
- Updated HL7 v2 to FHIR mappings
- Updated FHIR files in integrations tests

## Checklist

### Testing

## Linked Issues
- Fixes #7566
